tests/snapshot_tests.cc

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
```

Comment:
Perhaps you can move this into database_test.cc, the proliferation of test files increases build times. The tests there are sort of related.